server/metrics.go

```diff
@@ -199,4 +211,23 @@ func init() {
 	prometheus.MustRegister(apiConcurrencyGauge)
 	prometheus.MustRegister(forwardFailCounter)
 	prometheus.MustRegister(forwardTsoDuration)
+	prometheus.MustRegister(regionRequestCounter)
+}
+
+func incRegionRequestCounter(method string, header *pdpb.RequestHeader, err *pdpb.Error) {
```

Comment:
Hash function is little heavy, and we don't need to report all the request just see the the source. So we can use sample to report the normal request to decrease the usage of hash function.

Reply:
This can indeed lighten the load; for example, we can collect metrics with a probability of one percent. Even with tens of thousands of requests, we can still identify the sources of these large requests.

If there are no objections from others, I will make the changes today.

Reply:
Done~ https://github.com/tikv/pd/pull/9117/commits/7c00c9e9024d5236efbf4adb15d3aa8ace494eb7